python/taichi/lang/kernel_arguments.py

```diff
@@ -108,7 +108,7 @@ def decl_sparse_matrix(dtype, name):
 
 def decl_ndarray_arg(element_type, ndim, name, needs_grad, boundary):
     arg_id = impl.get_runtime().compiling_callable.insert_ndarray_param(element_type, ndim, name, needs_grad)
-    return AnyArray(_ti_core.make_external_tensor_expr(element_type, ndim, arg_id, needs_grad, boundary))
+    return AnyArray(_ti_core.make_external_tensor_expr(element_type, ndim, arg_id, needs_grad, boundary), element_type)
```

Comment:
`element_type` is already passed in as the first argument ;)

Reply:
However `_ti_core.make_external_tensor_expr` returns an expression, and we can't easily extract the `element_type` out of it.